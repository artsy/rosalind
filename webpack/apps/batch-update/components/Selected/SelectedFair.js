import React from 'react'
import SelectedComponent from './SelectedComponent'

const SelectedFair = (props) => {
  return (
    <div>
      <h2>Fair</h2>
      <SelectedComponent
        stateKey='fair'
        name={props.name}
        onRemove={props.clearState}
      />
    </div>
  )
}

export { SelectedFair }
