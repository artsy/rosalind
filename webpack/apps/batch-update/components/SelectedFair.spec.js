import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import SelectedFair from './SelectedFair'

let props

beforeEach(() => {
  props = {
    fair: {
      id: 'frieze',
      name: 'Frieze'
    },
    onClearFair: jest.fn()
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SelectedFair {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('fires the remove handler on click', () => {
  const wrapper = mount(<SelectedFair {...props} />)
  const mockClickEvent = { preventDefault: jest.fn() }
  wrapper.find('a').simulate('click', mockClickEvent)
  expect(props.onClearFair.mock.calls.length).toEqual(1)
})
