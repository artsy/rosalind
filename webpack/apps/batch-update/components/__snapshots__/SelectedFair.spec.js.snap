exports[`test renders correctly 1`] = `
<div
  className="crrNnH">
  Frieze
  <a
    className="remove"
    href="#"
    onClick={[Function]}>
    âœ•
  </a>
</div>
`;
